pkg/types/service_broker.go

```diff
@@ -77,6 +97,14 @@ func (e *ServiceBroker) transform(ctx context.Context, transformationFunc func(c
 	if err != nil {
 		return err
 	}
+
+	if e.Credentials.TLS != nil && e.Credentials.TLS.Key != "" {
+		transformedPrivateKey, err := transformationFunc(ctx, []byte(e.Credentials.TLS.Key))
+		if err != nil {
+			return err
+		}
+		e.Credentials.TLS.Key = string(transformedPrivateKey)
+	}
```

Comment:
move this section so it is not in the middle of password transformation

Reply:
I thing we would still rather keep it as is - as in this case it should be safer to return an empty TLS config to avoid problems going onwards, for instance when there're attempts to convert it to string

Reply:
There's a reason this section is there:
In case there're non-empty TLS settings, then and only then, we'll apply transformation on the private key ( we can no longer assume that the broker will always haver a user and password as it used to be).
Otherwise we don't want to save it in the database.
if we move this section outside the 'if' - it will not work properly!